### PR TITLE
core#4120 Reduce duplicates groups/mailings when copying Mailings

### PR DESCRIPTION
**Pull request by shaneonabike**
Overview
----------------------------------------
See [issue #4120](https://lab.civicrm.org/dev/core/-/issues/4120)

Before
----------------------------------------
Old mailings with repeated groups/mailings are copied over to a new mailing causes large numbers of replicated groups.

After
----------------------------------------
Reduces the number to unique valus

### CODE REVIEW

**Comment by colemanw** [comment]
@MegaphoneJon can you review this one?

**Comment by shaneonabike** [comment]
Hey there @MegaphoneJon @colemanw 

I did outline one scenario in the actual ticket where you can actually add additional _replicated_ groups while viewing the newsletter (draft) in another language. There is no restriction language wise (which in itself is a bug but that's another discussion). This is how our client was able to amp up the list with multiple groups I believe.

After you have done that once you just need to send it, clone it, rinse and repeat. You'll find the list will get big.

**Comment by shaneonabike** [comment]
@MegaphoneJon I modified my ticket to make it clearer how to reproduce [here](https://lab.civicrm.org/dev/core/-/issues/4120#replicate)

**Comment by larssandergreen** [comment]
I had a look at this.
`r-run` is good, solves the issue.
Code looks fine.

I'm not entirely convinced this is the way to resolve this issue though. If the problems comes about because of [#3558](https://lab.civicrm.org/dev/core/-/issues/3558), then maybe it makes more sense to fix that issue and then this won't come up? Or are there other ways to get duplicate groups on a mailing in the first place?

**Comment by MegaphoneJon** [comment]
I experienced this problem on a monolingual site.  I know that you have to copy old mailings to make it happen but besides that I'm not sure.  Thank you @larssandergreen for testing, I was getting so busy!

**Comment by larssandergreen** [comment]
@MegaphoneJon I think fine to merge this if we think it can occur for a number of upstream reasons and we don't think we can resolve them all.

**Comment by larssandergreen** [comment]
I think it's fine to merge this to at least make the situation better for now, doesn't hurt.

File: api/v3/Mailing.php [hunk 1]
```diff
@@ -200,7 +200,10 @@ function civicrm_api3_mailing_clone($params) {
   while ($dao->fetch()) {
     // CRM-11431; account for multi-lingual
     $entity = (substr($dao->entity_table, 0, 15) == 'civicrm_mailing') ? 'mailings' : 'groups';
-    $newParams[$entity][strtolower($dao->group_type)][] = $dao->entity_id;
+    $group_type = strtolower($dao->group_type);
+    if (!in_array($dao->entity_id, $newParams[$entity][$group_type])) {
```
Comment (aydun) [inline api/v3/Mailing.php:204]: `in_array()` without it's third parameter being `TRUE` can produce weird results so preference is always to specify it.

Comment (shaneonabike) [reply api/v3/Mailing.php]: Ok @aydun ! I don't know why I never received an email with your comment - or maybe it just got lost in the shuffle. I fixed this now in the latest.

**Pull request closed without merging**